File: src/main/java/no/nav/sosialhjelp/soknad/consumer/svarut/SvarUtService.java
package no.nav.sosialhjelp.soknad.consumer.svarut;

import no.ks.fiks.svarut.klient.model.Forsendelse;
import org.springframework.stereotype.Component;

import java.io.InputStream;
import java.util.Map;

@Component
public class SvarUtService {

    private final SvarUtConsumer svarUtConsumer;

    public SvarUtService(SvarUtConsumer svarUtConsumer) {
        this.svarUtConsumer = svarUtConsumer;
    }

    public String send(Forsendelse forsendelse, Map<String, InputStream> filnavnInputStreamMap) {
        var forsendelseId = svarUtConsumer.sendForsendelse(forsendelse, filnavnInputStreamMap);
        return forsendelseId.getId().toString();
    }

}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/innsending/svarut/client/SvarUtService.kt
package no.nav.sosialhjelp.soknad.innsending.svarut.client

import no.ks.fiks.svarut.klient.model.Forsendelse
import org.springframework.stereotype.Component
import java.io.InputStream

@Component
class SvarUtService(
    private val svarUtClient: SvarUtClient
) {

    fun send(forsendelse: Forsendelse, filnavnInputStreamMap: Map<String, InputStream>): String? {
        val forsendelseId = svarUtClient.sendForsendelse(forsendelse, filnavnInputStreamMap)
        return forsendelseId?.id.toString()
    }
}
